src/main/scala/magellan/Polygon.scala

```diff
@@ -169,6 +169,20 @@ class Polygon extends Shape {
     intersects
   }
 
+  /**
+    * A polygon wholly contains a line if none of its edges intersect the line
+     AND contains one of the line endpoints.
+    *
+    * @param line
+    * @return
+    */
+  private [magellan] def contains(line: Line): Boolean = {
```

Comment:
contains does not mean wholly contains... a polygon contains a line even if the edges are touching as long as the interior is fully contained. you probably want to introduce the concept of strict containment here

Reply:
I see. It is getting complicated

Reply:
would the equivalent PostGIS concept be ST_Covers?